crates/fluvio-storage/src/mut_index.rs

```diff
@@ -253,6 +258,12 @@ impl Index for MutLogIndex {
     }
 
     fn len(&self) -> Size {
+        self.first_empty_slot
+            .checked_mul(INDEX_ENTRY_SIZE)
+            .expect("index size must not exceed u32::MAX")
```

Comment:
we should not have expect in the non test code

Reply:
Done